nautobot/ipam/models.py

```diff
@@ -289,14 +333,16 @@ class Meta:
         unique_together = [
             ["vrf", "device"],
             ["vrf", "virtual_machine"],
+            ["vrf", "virtual_device_context"],
+            ["virtual_device_context", "rd", "name"],
```

Comment:
Changing my mind here due to the previous comment - let's leave this constraint as a TODO for now. I think if anything it should probably be two separate constraints, i.e `[vdc, rd]` and `[vdc, name]` but let's not try to solve that at the moment.